### ISSUE TITLE
Reports of Windows users losing peers

### ISSUE THREAD

**Issue by james-prysm** [prysmaticlabs/prysm#13431]
# 💎 Issue

### Background

Opening this as it's nuanced and haven't been wide spread however it does randomly happen to some windows users
where the OS decides to drop all packets from prysm for some reason. This results in a reduced peer count. Root cause is currently unknown but opening this as a tracking issue. 

related https://github.com/prysmaticlabs/documentation/issues/891
related https://github.com/prysmaticlabs/prysm/issues/8144

**Comment by defeedme** [comment]
welp my system worked fine for 3 years and now I lose all peers after about 2 hours, had to setup a schedule task to kill and restart beacon every 2 hours.. sucks. version 4.2 hasn't fixed anything for windows

**Comment by defeedme** [comment]
Hi all, it appears the problem was with my Time Sync program (net time) .. I had the sync interval set at 2 hours.. the ntp servers must have been giving bad data or something.. and windows sucks with accurate time to begin with.. So the solution was to use the google time servers and change it to 10 min sync interval.. so far so good!!! 2 days almost perfect!

**Comment by james-prysm** [comment]
i'll try to add this to the docs

**Comment by james-prysm** [comment]
closing issue with added docs now

**Comment by defeedme** [comment]
welp guess i spoke to soon. back to the same problem. it worked for 2-3 days then stopped. I rebooted my router / modem and same thing worked for a day.. I'm really at my wits end.  can't understand how when i stop and restart beacon it works for a few hours then stops. its maddening.

**Comment by web3dev2023** [comment]
Same issue. I have a Windows 10 and a Windows 11 PC. The beacon client running on them loses peers after a few hours. I sync the Windows time every 5 minutes but it does not work. 

They had been working fine for months. I guess the issue became prominent since the Dencun upgrade.

**Comment by defeedme** [comment]
> Same issue. I have a Windows 10 and a Windows 11 PC. The beacon client running on them loses peers after a few hours. I sync the Windows time every 5 minutes but it does not work.
> 
> They had been working fine for months. I guess the issue became prominent since the Dencun upgrade.

following.. please let me know if you find out anything.. It seems like something they did around December really made it bad for windows users..   I would love to switch to a minority client but just haven't had the time to figure it out..

**Comment by web3dev2023** [comment]
> let me know if you find out anything.. It seems like something they did around December really made it bad for windows users.. I would love to switch to a minority client but just haven't had the time to figure it out..
> 
> can you add your comment to this new issue


Yes, I did found a workaround:

- I am now using the lighthouse beacon client and it has no issue in finding peers.

- At the same time geth can find a few peers before losing them and I added them to static peers:
1. First generate a config file: `geth.exe --mainnet dumpconfig > geth-config.toml`
2. Then edit this file, add your favourite peers to `StaticNodes` , save it with `UTF-8` encoding (I used Notepad++)
3. Finally run geth with the flag `--config geth-config.toml`

**Comment by defeedme** [comment]
> > let me know if you find out anything.. It seems like something they did around December really made it bad for windows users.. I would love to switch to a minority client but just haven't had the time to figure it out..
> > can you add your comment to this new issue
> 
> Yes, I did found a workaround:
> 
> * I am now using the lighthouse beacon client and it has no issue in finding peers.
> * At the same time geth can find a few peers before losing them and I added them to static peers:
> 
> 1. First generate a config file: `geth.exe --mainnet dumpconfig > geth-config.toml`
> 2. Then edit this file, add your favourite peers to `StaticNodes` , save it with `UTF-8` encoding (I used Notepad++)
> 3. Finally run geth with the flag `--config geth-config.toml`

thanks so much for the workaround! i've been suffering for months lol.. at least now I know I'm not crazy and this is definitely a real issue.. lighthouse here we come

**Comment by defeedme** [comment]
> 1. First generate a config file: `geth.exe --mainnet dumpconfig > geth-config.toml`
> 2. Then edit this file, add your favourite peers to `StaticNodes` , save it with `UTF-8` encoding (I used Notepad++)
> 3. Finally run geth with the flag `--config geth-config.toml`

i'm very confused because I'm using nethermind (i gave up on geth).. can you give me specific steps on how to migrate from prysm to lighthouse using nethermind?  thanks!

**Comment by web3dev2023** [comment]
> > 1. First generate a config file: `geth.exe --mainnet dumpconfig > geth-config.toml`
> > 2. Then edit this file, add your favourite peers to `StaticNodes` , save it with `UTF-8` encoding (I used Notepad++)
> > 3. Finally run geth with the flag `--config geth-config.toml`
> 
> i'm very confused because I'm using nethermind (i gave up on geth).. can you give me specific steps on how to migrate from prysm to lighthouse using nethermind? thanks!

It's basically the same as running prysm, here is my script. You might need to sync from genesis.
`C:\Users\admin\lighthouse.exe bn --network mainnet --execution-endpoint http://localhost:8551 --datadir d:/ethereum/execution --execution-jwt D:/ethereum/consensus/prysm/jwt.hex --target-peers 60`

**Comment by theandygould** [comment]
I'm having the same issues and it's so frustrating. I'm also not a developer/programmer so I'm pretty nooby when it comes to this stuff. Is there more documentation on installing lighthouse for windows? I'm trying to download their pre-built binary for windows but keep getting error message when I try to run lighthouse
![Screenshot (9)](https://github.com/prysmaticlabs/prysm/assets/79128132/88f27568-71cf-49cd-924b-3b9c628a9a87)

**Comment by VirtualLands** [comment]
I've been tried to sync a full node for more than three weeks. 
I run "Nethermind 1.25.0 + Prysm v5.0.0" on a platform "windows 11 + docker desktop 4.31.0". Prysm peer numbers would reduces to 0 after about 2 hours. To be exact, 2 hours after "Blob filesystem cache warm-up complete".  It may take 10 to 40 minute to reach "Blob filesystem cache warm-up complete", then two hours later I would loss all peers suddenly.
I tried 'time server intervals', 'Configure ports and firewalls for improved peer-to-peer connectivity', I got the same results.

Then I tried I run "Nethermind 1.27.0 + Prysm v5.0.4" on a platform "windows 11 + docker desktop 4.31.1".  I set --max-concurrent-dials=10, 5, or 1.  Same results.

Today, I installed the 'Nethermind 1.27.0' and 'Prysm v5.0.4' on windows directly. The command is as below.
./prysm --mainnet --p2p-tcp-port=9000 --p2p-udp-port=9000 --jwt-secret=D:\Ethereum\jwtsecret --rpc-host=0.0.0.0 --rpc-port=4001 --grpc-gateway-host=0.0.0.0 --grpc-gateway-port=4000 --datadir=D:\Ethereum\Prysm  --verbosity=info --p2p-max-peers=50 --execution-endpoint=http://0.0.0.0:8551 --accept-terms-of-use --monitoring-host=0.0.0.0 --monitoring-port=5054 --checkpoint-sync-url=https://beaconstate.info --genesis-beacon-api-url=https://beaconstate.info --max-concurrent-dials=10

./nethermind --config=mainnet --datadir=D:\Ethereum\Nethermind --log=INFO --JsonRpc.Enabled=true --JsonRpc.Host=0.0.0.0 --JsonRpc.Port=8545 --JsonRpc.JwtSecretFile=D:\Ethereum\jwtsecret --JsonRpc.EngineHost=0.0.0.0 --JsonRpc.EnginePort=8551 --Network.DiscoveryPort=30303 --HealthChecks.Enabled=true --Metrics.Enabled=true --Metrics.ExposePort=8008

Everything is ok now.  It has been running smoothly for 12 hours.  And the network speed is ten times faster than before.   The network speed is 1-10M for docker desktop, and 100-200M for windows.

Hope there are some clues from my report.
